internal/daemon/controller/handlers/workers/worker_service.go

```diff
@@ -393,3 +473,22 @@ func validateListRequest(req *pbs.ListWorkersRequest) error {
 func validateDeleteRequest(req *pbs.DeleteWorkerRequest) error {
 	return handlers.ValidateDeleteRequest(handlers.NoopValidatorFn, req, servers.WorkerPrefix)
 }
+
+func validateUpdateRequest(req *pbs.UpdateWorkerRequest) error {
+	return handlers.ValidateUpdateRequest(req, req.GetItem(), func() map[string]string {
+		badFields := map[string]string{}
+		if req.GetItem().GetWorkerConfig() != nil {
+			badFields[globals.ConfigurationField] = "This is a read only field."
+		}
+		if req.GetItem().GetCanonicalAddress() != "" {
+			badFields[globals.CanonicalAddressField] = "This is a read only field."
```

Comment:
perhaps add unit tests for badFields for: CanonicalAddress, CanonicalTags and Tags?